packages/enzyme/src/ShallowWrapper.js

```diff
@@ -399,11 +399,11 @@ class ShallowWrapper {
    * @param {Function} cb - callback function
    * @returns {ShallowWrapper}
    */
-  setProps(props, callback = noop) {
+  setProps(props, callback) {
```

Comment:
either way, the default is required, so the `.length` of the function is correct.